src/api/apiErrors.js

```diff
@@ -0,0 +1,57 @@
+/* @flow strict-local */
+import type { ApiErrorCode, ApiResponseErrorData } from './transportTypes';
+import * as logging from '../utils/logging';
+
+/** Runtime class of custom API error types. */
+export class ApiError extends Error {
+  code: ApiErrorCode;
+  data: $ReadOnly<{ ... }>;
+  httpStatus: number;
+
+  constructor(httpStatus: number, data: $ReadOnly<ApiResponseErrorData>) {
+    // eslint-disable-next-line no-unused-vars
+    const { result, code, msg, ...rest } = data;
+    super(msg);
+    this.data = rest;
+    this.code = code;
+    this.httpStatus = httpStatus;
```

Comment:
Picky nit: I find it's slightly easier to read when a series of properties (etc.) comes in the same order in different places -- makes it a bit less work to look back and forth between them and compare.

Here, that'd mean initializing these three in the same order they're declared above.